tests/ServiceResolver/Resolvers/MemoryCacheTest.php

```diff
@@ -23,35 +25,34 @@ protected function setUp(): void
      */
     public function forwardCallToDecoratedResolver(): void
     {
-        $serviceDefinition = ServiceDefinition::build(fn() => 'fake');
+        $actualDefinition = $this->createMock(Definition::class);
         $this->decorated
             ->expects($this->once())
             ->method('get')
             ->with('some-id')
-            ->willReturn($serviceDefinition);
+            ->willReturn($actualDefinition);
 
         $result = $this->memoryCache->get('some-id');
 
-        $this->assertSame($serviceDefinition, $result);
+        $expected = new CachedDefinition($actualDefinition);
```

Comment:
This is leaking implementation details about `MemoryCache`, right? Not necessarily bad but something that must be on your radar.

Reply:
This is a good question. This is covered anyway by the `ServiceResolverTest`, but wouldn't it be important to ensure `MemoryCache` is decorating the result properly?

Reply:
I took a look again and the problem here is the nature of the test itself. First of all, the behavior is actually being tested on `returnTheSameServiceIfItWasAlreadyCalledBefore` of the `ServiceResolverTest` which ensures the same instance is returned (that is what you suggested). That means the `forwardCallToDecoratedResolver` of `MemoryCacheTest` is a redundant test.

So, I will keep the test the way for now. It might be a bad idea, but in this case we will discover it in the future.